3.workload.md

```diff
@@ -0,0 +1,378 @@
+# 3. Workload
+
+This section defines the workload definition schematics.
+
+Workload definitions are provided by the platform (just as traits are) so that users may inspect the platform and
+ learn what workloads are available for use. Workload definitions are not extensible to platform users (only to platform
+  operators). Thus, platform users MUST NOT be allowed to create new workload kinds.
+
+## Top-Level Attributes
+
+The following attributes are common across all schemata defined in this document. This structure is designed to allow
+ object type to be determined by scanning two fields (`apiVersion` and `kind`), and provide access to common fields
+  in the `metadata` section. The `spec` section is where type-specific information is located.
+
+Here are the attributes that provide top-level information about the workload definition.
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `apiVersion` | `string` | Y | | The specific version of the specification in use. The core types use `core.oam.dev/v1alpha2` |
+| `kind` | `string` | Y || The string `WorkloadDefinition` |
+| `metadata` | [`Metadata`](#metadata) | Y | | WorkloadDefinition metadata. |
+| `spec`| [`Spec`](#spec) | Y | | A container for the workload definition. |
+
+### Metadata
+
+The metadata describes this workload definition. The `metadata` section is defined in [the meta section of part 2](2.overview_and_terminology.md#Metadata)
+
+### Spec
+
+The specification section contains a reference to the workload schema
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `DefinitionRef` | [`DefinitionRef`](#DefinitionRef) | Y | | Workload schema reference. |
+
+
+#### DefinitionRef
+
+DefinitionRef defines where to find the workload schema
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `name` | `string` | Y | | The workload schema's name. Must be formatted in a way that can uniquely identify a specific version of the definition  |
```

Comment:
We really need sample to make spec easy to understand.

Also I'm not sure that "indexing" works for non-CRD schema. I'm thinking maybe:

```yaml
definitionRef:
  name: scaling.alibaba.com/v1alpha2.Manualscaling
  schematic: "json schema string" # this is a optional field only needed for non-CRD platforms
```

Reply:
> Must be formatted in a way that can uniquely identify a specific version of the definition

Cross-linking this discussion; https://github.com/crossplane/crossplane/pull/1276#pullrequestreview-362313530. I'm really not sure requiring a specific version makes sense in the Kubernetes world.

Reply:
would it make sense in a non-k8s world?

Reply:
I am in favor of Ryan's proposal:

```yaml
name: containerisedworkload.core.oam.dev
version: v1beta1
```
The version field should be optional of course (only needed when multiple version CRDs are installed which is a rare case).

Reply:
> Also I'm not sure that "indexing" works for non-CRD schema. I'm thinking maybe:

it seems that it should work as long as the "name" is unique? Is it hard to do? With the schematic and name together, what does the "name" mean here?